config/dependency.go

```diff
@@ -891,6 +891,18 @@ func runTerragruntOutputJson(ctx *ParsingContext, targetConfig string) ([]byte,
 	stdoutBufferWriter := bufio.NewWriter(&stdoutBuffer)
 	ctx.TerragruntOptions.Writer = stdoutBufferWriter
 
+	currentTerraformLogsToJson := ctx.TerragruntOptions.TerraformLogsToJson
+	currentIncludeModulePrefix := ctx.TerragruntOptions.IncludeModulePrefix
+
+	// reset the settings after the run
+	defer func() {
+		ctx.TerragruntOptions.TerraformLogsToJson = currentTerraformLogsToJson
+		ctx.TerragruntOptions.IncludeModulePrefix = currentIncludeModulePrefix
+	}()
+	// Disable json wrapping to allow reading of outputs
+	ctx.TerragruntOptions.TerraformLogsToJson = false
+	ctx.TerragruntOptions.IncludeModulePrefix = false
```

Comment:
```suggestion
	newOpts := *ctx.TerragruntOptions
	newOpts.TerraformLogsToJson = false
	newOpts.IncludeModulePrefix = false
	newOpts.Writer = stdoutBufferWriter

	ctx = ctx.WithTerragruntOptions(&newOpts)
```

Reply:
I think it's better to create a new context. it will be more reliable with concurrency.